File: code/xml/javascript/XMLSerializer.js
const fs = require('fs');
const { promisify } = require('util');
const xmlbuilder = require('xmlbuilder');

const writeFileAsync = promisify(fs.writeFile);

// Objeto Customer para serialização
const customer = {
    id: 1,
    name: 'John Doe',
    age: 30
};

// Construindo o XML
const xml = xmlbuilder.create({ customer }).end({ pretty: true });

// Escrevendo o XML em um arquivo
writeFileAsync('customer.xml', xml)
    .then(() => console.log('XML serialized successfully'))
    .catch(err => console.error('Error serializing XML:', err));

File: code/xml/python/XMLSerializer.py
import xml.etree.ElementTree as ET

# Criando elementos XML para o objeto Customer
customer = ET.Element("customer")
id = ET.SubElement(customer, "id")
id.text = "1"
name = ET.SubElement(customer, "name")
name.text = "John Doe"
age = ET.SubElement(customer, "age")
age.text = "30"

# Criando a árvore XML
tree = ET.ElementTree(customer)
# Escrevendo a árvore XML em um arquivo
tree.write("customer.xml", encoding="utf-8", xml_declaration=True)
